opentelemetry/src/global/error_handler.rs

```diff
@@ -47,40 +43,3 @@ impl<T> From<PoisonError<T>> for Error {
         Error::Other(err.to_string())
     }
 }
-
-struct ErrorHandler(Box<dyn Fn(Error) + Send + Sync>);
```

Comment:
I'd suggest to split the PR into smaller ones - 
1. Review and replace global handler with error! internal logs, one crate or even small scope in one PR
2. When we are done with all crates, removing the global handler can be a simple PR.
3. Revisit each internal log - if it is not user actionable, and only good for sdk owners, then it should be debug level.